core/domain/topic_domain_test.py

```diff
@@ -821,6 +821,497 @@ def test_cannot_move_existing_skill_to_subtopic(self) -> None:
+        an exception is raised. Also checks for the sub topic
+        getting added to the topic.
+        """
+        incorrect_new_subtopic_id = 3
+        correct_new_subtopic_id = 2
+        expected_subtopic_id = self.topic.next_subtopic_id
+        with self.assertRaisesRegex(
+            Exception,
+            'The given new subtopic id %s is not equal to the expected next '
+            'subtopic id: %s' % (
+                incorrect_new_subtopic_id,
+                expected_subtopic_id
+            )
+        ):
+            self.topic.add_subtopic(
+                incorrect_new_subtopic_id,
+                'subtopic_3',
+                'url_frag'
+            )
+        self.topic.add_subtopic(
+            correct_new_subtopic_id,
+            'subtopic_title',
+            'url_frag'
+            )
+        self.assertEqual(2, len(self.topic.subtopics))
+
+    def test_update_practice_tab_is_displayed(self) -> None:
+        self.topic.update_practice_tab_is_displayed(True)
+        self.assertTrue(self.topic.practice_tab_is_displayed)
+
+    def test_update_page_title_fragment_for_web(self) -> None:
+        updated_frag = 'updated fragment'
+        self.topic.update_page_title_fragment_for_web(updated_frag)
+        self.assertEqual(self.topic.page_title_fragment_for_web, updated_frag)
+
+    def test_update_meta_tag_content(self) -> None:
+        updated_meta_tag = 'updated meta tag'
+        self.topic.update_meta_tag_content(updated_meta_tag)
+        self.assertEqual(self.topic.meta_tag_content, updated_meta_tag)
+
+    def test_update_description(self) -> None:
+        updated_desc = 'updated description'
+        self.topic.update_description(updated_desc)
+        self.assertEqual(self.topic.description, updated_desc)
+
+    def test_update_thumbnail_file_and_size(self) -> None:
+        updated_file_name = 'file_name.svg'
+        updated_size = 1234
+        self.topic.update_thumbnail_filename_and_size(
+            updated_file_name,
+            updated_size
+        )
+        self.assertEqual(self.topic.thumbnail_filename, updated_file_name)
+        self.assertEqual(self.topic.thumbnail_size_in_bytes, updated_size)
+
+    def test_update_url_fragment(self) -> None:
+        url_frag = 'url fragment'
+        self.topic.update_url_fragment(url_frag)
+        self.assertEqual(self.topic.url_fragment, url_frag)
+
+    def test_update_name(self) -> None:
+        updated_name = 'updated name'
+        self.topic.update_name(updated_name)
+        self.assertEqual(self.topic.name, updated_name)
+
+    def test_update_name_bytes(self) -> None:
+        updated_name = b'updated name'
+        with self.assertRaisesRegex(
+            utils.ValidationError,
+            'Name should be a string.'
+        ):
+            # Here we use MyPy ignore because we need to pass
+            # a non-string to test the full function.
+            self.topic.update_name(updated_name) # type: ignore
```

Comment:
Hmm I think this indicates that the type on `update_name` is wrong. If the function can handle non-strings and raise validation errors, then it's type shouldn't be `str`. @vojtechjelinek wdyt?

Reply:
Ah, this is a bit of an exception.

@jonlake5 Please do this
```suggestion
            # TODO(#13059): Here we use MyPy ignore because after we fully type
            # the codebase we plan to get rid of the tests that intentionally
            # test wrong inputs that we can normally catch by typing.
            self.topic.update_name(updated_name) # type: ignore[assignment]
```

Reply:
Had to use arg-type for ignore otherwise mypy complained of an unused ignore.
`error: unused 'type: ignore' comment`
Done.
`self.topic.update_name(updated_name) # type: ignore[arg-type]`